Dynatrace/dynatrace-operator
# Use kind "Pod" when owner reference is empty

# Description

When the install oneagent container is injected, the object kind of the owner of a pod is used to set the workload kind env variable for the standalone command. If the there is now owner reference and the kind is a pod, the kind is set to an empty string.
The standalone runner interprets this empty string as a missing env var and errors out.

This change removes setting the kind to an empty string, which means in the scenario above the workload kind will be set to "Pod", fixing this issue

## How can this be tested?
* Deploy the operator in either cloud native or application monitoring
* Deploy secret, dynakube, etc.
* Run a single pod without using a deployment, e.g., `kubectl run ...`
* Check that the new pod is injected into correctly and the install-oneagent container succeeds

## Checklist
